django_dbconn_retry/__init__.py

```diff
@@ -53,15 +56,19 @@ def ensure_connection_with_retries(self: django_db_base.BaseDatabaseWrapper) ->
                     self.connect()
                 except Exception as e:
                     if isinstance(e, _operror_types):
-                        if hasattr(self, "_connection_retries") and self._connection_retries >= 1:
+                        if hasattr(self, "_connection_retries") and self._connection_retries >= MAX_DBCONN_RETRY_TIMES:
                             _log.error("Reconnecting to the database didn't help %s", str(e))
                             del self._in_connecting
                             post_reconnect.send(self.__class__, dbwrapper=self)
                             raise
                         else:
                             _log.info("Database connection failed. Refreshing...")
                             # mark the retry
-                            self._connection_retries = 1
+
+                            if not hasattr(self, "_connection_retries"):
+                                self._connection_retries = 1
+                            else:
+                                self._connection_retries += 1
```

Comment:
Following EAFP, this might be a suitable solution too.
```suggestion
                            try:
                                self._connection_retries += 1
                            except AttributeError:
                                self._connection_retries = 1
```